chore(bld): update for newer metatomic-torch

HaoZeke opened the pull request:
also uses torch 2.9 for newer LAMMPS

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* ~[ ] Reset the build number to `0` (if the version changed)~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

HaoZeke committed:
Update to the latest dependencies for metatensor & co.

HaoZeke commented:
@Luthaf let's squash merge this ASAP, wanna unblock CI over at https://github.com/lab-cosmo/atomistic-cookbook/pull/167

I think it needs a companion PR to the `py-plumed` recipe too

Luthaf commented on recipe/meta.yaml:
```diff
@@ -60,16 +60,15 @@ requirements:
     # always build against the CPU version of libtorch, we can still pick the
     # cuda one at runtime
     - libtorch * cpu*
-    - libtorch
```
Please leave this one around, this is how conda-smithy tracks a dependency on libtorch

Luthaf replied:
This will likely need a rerender after adding it back

HaoZeke replied:
weirdly didn't need one, but does break

Luthaf committed:
Rebuild for pytorch v2.9

Luthaf committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.02.23.00.48

Other tools:
- conda-build 25.11.0
- rattler-build 0.51.0
- rattler-build-conda-compat 1.4.10

HaoZeke commented:
Still needs some work

```
nox > {
  "allow_retry": false,
  "caused_by": "None",
  "error": "LibMambaUnsatisfiableError: Encountered problems while solving:\n  - package lammps-metatomic-2025.09.10.mta0-cuda126_kokkos_arch_VOLTA70_hd77f0b9_nompi_git.e9b222e_0 requires libmetatensor-torch >=0.8.1,<0.9.0, but none of the providers can be installed\n\nCould not solve for environment specs\nThe following packages are incompatible\n\u251c\u2500 lammps-metatomic =* * is not installable because there are no viable options\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta0 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.1,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* nompi* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0 would require\n\u2502  \u2502     \u2502  \u2514\u2500 libmetatensor-torch >=0.7.6,<0.8.0 *, which conflicts with any installable versions previously reported;\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta0 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.1,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* mpi_mpich* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta0 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.1,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* mpi_openmpi* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta1 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.2,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* mpi_mpich* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta1 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.2,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* mpi_openmpi* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u2514\u2500 lammps-metatomic 2025.09.10.mta1 would require\n\u2502     \u251c\u2500 libmetatensor-torch >=0.8.2,<0.9.0 *, which can be installed;\n\u2502     \u2514\u2500 plumed-metatomic =* nompi* but there are no viable options\n\u2502        \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502        \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2514\u2500 py-plumed-metatomic =* * is not installable because it requires\n   \u2514\u2500 plumed-metatomic =2.10.0.dev0 *, which cannot be installed (as previously explained).",
  "exception_name": "LibMambaUnsatisfiableError",
  "exception_type": "<class 'conda_libmamba_solver.exceptions.LibMambaUnsatisfiableError'>",
  "message": "Encountered problems while solving:\n  - package lammps-metatomic-2025.09.10.mta0-cuda126_kokkos_arch_VOLTA70_hd77f0b9_nompi_git.e9b222e_0 requires libmetatensor-torch >=0.8.1,<0.9.0, but none of the providers can be installed\n\nCould not solve for environment specs\nThe following packages are incompatible\n\u251c\u2500 lammps-metatomic =* * is not installable because there are no viable options\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta0 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.1,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* nompi* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0 would require\n\u2502  \u2502     \u2502  \u2514\u2500 libmetatensor-torch >=0.7.6,<0.8.0 *, which conflicts with any installable versions previously reported;\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta0 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.1,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* mpi_mpich* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta0 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.1,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* mpi_openmpi* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta1 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.2,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* mpi_mpich* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u251c\u2500 lammps-metatomic 2025.09.10.mta1 would require\n\u2502  \u2502  \u251c\u2500 libmetatensor-torch >=0.8.2,<0.9.0 *, which can be installed;\n\u2502  \u2502  \u2514\u2500 plumed-metatomic =* mpi_openmpi* but there are no viable options\n\u2502  \u2502     \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502  \u2502     \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2502  \u2514\u2500 lammps-metatomic 2025.09.10.mta1 would require\n\u2502     \u251c\u2500 libmetatensor-torch >=0.8.2,<0.9.0 *, which can be installed;\n\u2502     \u2514\u2500 plumed-metatomic =* nompi* but there are no viable options\n\u2502        \u251c\u2500 plumed-metatomic 2.10.0.dev0, which cannot be installed (as previously explained);\n\u2502        \u2514\u2500 plumed-metatomic [2.10.0|2.10.0.dev1] conflicts with any installable versions previously reported;\n\u2514\u2500 py-plumed-metatomic =* * is not installable because it requires\n   \u2514\u2500 plumed-metatomic =2.10.0.dev0 *, which cannot be installed (as previously explained)."
}
```

Which as far as I can tell is that `libmetatensor-torch` needed an update, and `py-plumed-metatomic`
